docs/production/upgrade-or-modify.md

```diff
@@ -102,6 +102,43 @@ git_repo_url = https://github.com/zulip/zulip.git
 See also our documentation on [upgrading
 docker-zulip](https://github.com/zulip/docker-zulip#upgrading-from-a-git-repository).
 
+## Updating configuration files
```

Comment:
Is this the right place to put this?  I'm not sure if it's important to have "troubleshooting" more close to the actual upgrade instructions.  I guess we already link to there from the non-Git instructions, and upgrading from Git is relatively rare.

Another theory is that we could put this just before "Upgrading the operating system", and then link to it from https://zulip.readthedocs.io/en/latest/overview/changelog.html#upgrade-notes (which we encourage folks to read whenever looking at doing an upgrade).

This is a step that one only needs to do occasionally, and it's OK if folks forget; we're basically just providing updated inline documentation.

Reply:
I think we can try this out and iterate from here; it's not hard to backport documentation changes if we decide to move this.

(I do worry a bit that the heading here should be instead "Updating settings.py comment documentation" -- so that it's clear from scanning the headings that nothing bad will happen if you don't update the configuration files atomically with the update itself.

Reply:
Yeah.  I think the other heading which is confusing here is "Preserving local changes to configuration files" which is referring to things like nginx and postgres, but sits weirdly with "Updating configuration files" right near it.

Reply:
Reworded both of these headings.